File: C/Algorithms/746.c
int minCostClimbingStairs(int *cost, int costSize) {
  int tableSize = costSize + 2;
  int *table = malloc(tableSize * sizeof(*table));
  table[0] = 0;
  for (int i = 1; i < tableSize; i++)
    table[i] = 2 << 20;
  for (int i = 1; i < tableSize; i++) {
    int previous = table[i - 1];
    int cand_1 = previous + (i < costSize ? cost[i - 1] : 0);
    if (cand_1 < table[i])
      table[i] = cand_1;

    int j = i + 1;
    if (j < tableSize) {
      cand_1 = previous + (i < costSize ? cost[i] : 0);
      if (cand_1 < table[j])
        table[j] = cand_1;
    }
  }
  int res = table[tableSize - 1];
  free(table);
  return res;
}

File: Rust/Algorithms/746.rs
use std::cmp::min;

impl Solution {
    pub fn min_cost_climbing_stairs(cost: Vec<i32>) -> i32 {
        let mut table = vec![2 << 20; cost.len() + 2];
        table[0] = 0;

        for i in 1..table.len() {
            let prev = table[i - 1];
            table[i] = min(
                table[i],
                prev + if (i < cost.len()) { cost[i - 1] } else { 0 },
            );
            let j = i + 1;
            if (j < table.len()) {
                table[j] = min(
                    table[j],
                    prev + if (i < cost.len()) { cost[i] } else { 0 },
                );
            }
        }
        // return final element.
        table[table.len() - 1]
    }
}
